### ISSUE TITLE
Docs: add SAM support in examples folder

### ISSUE THREAD

**Issue by saragerion** [aws-powertools/powertools-lambda-typescript#606]
## Description of the feature request

Add examples of usage of the utilities in the `examples` folder.

**Problem statement**

**Summary of the feature**

- Infrastructure
- Lambda handler(s) code in TypeScript
- Readme with deployment guide 
- References to this sample in the project documentation

**Code examples**

CDK example for reference:
https://github.com/awslabs/aws-lambda-powertools-typescript/tree/main/examples/cdk

**Benefits for you and the wider AWS community**

Developers who use SAM can get started more quickly

**Describe alternatives you've considered**

Other IaC tool, which can be added later on in the roadmap.

### Related issues, RFCs

[docs(all): add CDK, Terraform, Serverless Framework, Terraform, Amplify examples](https://github.com/awslabs/aws-lambda-powertools-typescript/issues/385)  
[docs(examples): CDK example improvement with best practices](https://github.com/awslabs/aws-lambda-powertools-typescript/issues/520)

**Comment by dreamorosi** [comment]
Thank you Sara for opening the issue, as soon as we have a resolver please ask them know if they could leave a comment here so we can assign the issue & mark it as "in progress".

**Comment by bpauwels** [comment]
I will be picking up this issue

**Comment by flochaz** [comment]
I gave it a try few weeks ago if it can help : https://github.com/flochaz/powertools-javascript-sam-sample

**Comment by bpauwels** [comment]
I have just created the following PR: https://github.com/awslabs/aws-lambda-powertools-typescript/pull/674

As a reference, I used the API example that comes with AWS SAM. I have translated it into TypeScript and added the Powertools-Features to it.

**Comment by dreamorosi** [comment]
Before closing this issue and **after #674** has been merged, we need another PR to apply linting, formatting, etc. to the new examples folder. Change needed in:
- `lerna.json`
- `.github/workflows/*`

This way the new files will be picked up by the existing CI/CD steps.

**Comment by dreamorosi** [comment]
In #849 we have added the new SAM samples to the GitHub workflows so that these samples are listed, basic tested, and built as part of the CI/CD workflow.
